library/entropy.c

```diff
@@ -576,10 +578,12 @@ int mbedtls_entropy_source_self_test( int verbose )
 int mbedtls_entropy_self_test( int verbose )
 {
     int ret = 1;
+#if !defined(MBEDTLS_TEST_NULL_ENTROPY)
```

Comment:
Adding these macros works, but also makes the code difficult to read. An alternative could be to  remove all these preprocessor directives and after the `if` statement in line https://github.com/ARMmbed/mbedtls/pull/617/files#diff-2b8685bf4b9e38da9f8d54021cdd15dbR588 we could add the following three line:
```
#if !defined(MBEDTLS_TEST_NULL_ENTROPY)
    goto exit;
#endif
```
Clearly, we would also need to add the `exit` label in line https://github.com/ARMmbed/mbedtls/pull/617/files#diff-2b8685bf4b9e38da9f8d54021cdd15dbR641. I prefer the `goto` approach, but to be fair I am not sure either is actually very readable, so I will still upvote this one.